.gitignore

```diff
@@ -44,3 +44,6 @@ docs/build/
 # VSCode / clangd IntelliSense
 .vscode/
 .cache/
+
+# CLion / IDEA
+.idea/
```

Comment:
This should be in a local gitignore rather than in that of the project. However, we already have other settings for vscode, let's clean this in a dedicated PR.